safehttp/plugins/hsts/hsts.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package hsts
+
+import (
+	"strconv"
+	"strings"
+
+	"github.com/google/go-safeweb/safehttp"
+)
+
+// Plugin implements automatic HSTS functionality.
+type Plugin struct {
+	maxAge            uint64
+	includeSubDomains bool
+	preload           bool
+}
+
+// NewPlugin creates a new HSTS plugin with safe defaults.
+func NewPlugin() Plugin {
+	return Plugin{
+		maxAge:            63072000, // Two years in seconds.
+		includeSubDomains: true,
+		preload:           false,
```

Comment:
Why?

Reply:
If `preload` is sent from the server it can be registered in the browser preload list (https://hstspreload.org/). This causes all(/most) browsers to only ever send HTTPS requests to your server. It can take a long time to be removed from this list and if you don't want it or your server isn't prepared for it then that is bad. Therefore it is disabled by default. It is good to transition to having preload enabled eventually, but this can't be a default setting.